plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Debug/Internal.hs

```diff
@@ -345,6 +347,9 @@ throwingDischarged
     -> CekM uni fun s x
 throwingDischarged l t = throwingWithCause l t . Just . dischargeCekValue
 
+-- | Duplicate of 'Cek.Internal.runCekM'.
+--
+-- Exporting and reusing that function resulted in +20% perf loss for Cek, see #5223.
 runCekM
```

Comment:
I could also edit the existing `runCekM` to make use of `giveCekReqs` but I thought to keep it the same as the original cek internal module.

Reply:
Does `{-# INLINE runCekM #-}` help? It could be due to inlining vs. not, but could also be for other reasons.